### PR TITLE
Create separated deployment script and move to a BDD test including deploy.

### PR DESCRIPTION
**Pull request by michael-paddle**
In this PR I have separated out the actual deployment script from the testcase which deploys and tests.  I also rewrote this to use python behave as the test drive so that I can actually do an full ansible call during testing, checking the deploy script works as designed.

### CODE REVIEW

File: shippable.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 language: python
 
 python:
-  - 3.6
+  - 3.7
```
Comment (ghost) [inline shippable.yaml:5]: should this python version be the same as the one defined in your `deploy-sg-lambda.yml:93`?

File: deploy-sg-lambda.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            description: security group for testing cloudflare ip lambda
+            <<: *aws_connection_info
+          register: group_create
+
+        - name: test state=present - upload the lambda
+          lambda:
+            name: "{{lambda_function_name}}"
+            runtime: "python3.6"
+            handler: "{{lambda_handler}}"
+            role: "arn:aws:iam::455225740156:role/service-role/basicLambdaRole"
```
Comment (paddledave) [inline deploy-sg-lambda.yml:95]: Hardcoded account ID? Be nice to have as a param, even though just for testing

**Review by paddledave — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: deploy-sg-lambda.yml (hunk 2 above)
Comment (michael-paddle) [reply deploy-sg-lambda.yml:95]: It's overridable / overridden in deploy, but I will make this look nicer and add IAM role setup to the repo somehow.

File: shippable.yaml (hunk 1 above)
Comment (michael-paddle) [reply shippable.yaml:5]: Currently the test cases require 3.7 but amazon lambda is only providing 3.6 unfortunately;  thus the disconnect.

**Pull request merged**